transport/http2_server.go

```diff
@@ -163,6 +165,11 @@ func (t *http2Server) operateHeaders(hDec *hpackDecoder, s *Stream, frame header
 		return s
 	}
 	t.mu.Lock()
+	if t.goaway {
+		//Stop creating streams on this transport
```

Comment:
just something like

log.Printf("transport: http2Server has received GOAWAY and started to reject new streams.")